docs/design/services/dynamodb/high-level-library/prototype/option-2/sync/Prototype.java

```diff
@@ -0,0 +1,1077 @@
+/**
+ * A synchronous client for interacting (generically) with document databases.
+ *
+ * Features:
+ * <ol>
+ *     <li>Support for Java-specific types, like {@link Instant} and {@link BigDecimal}.</li>
+ *     <li>Support for reading and writing custom objects (eg. Java Beans, POJOs).</li>
+ * </ol>
+ *
+ * All {@link DocumentClient}s should be closed via {@link #close()}.
+ */
+@ThreadSafe
+public interface DocumentClient extends SdkAutoCloseable {
+    /**
+     * Create a {@link DocumentClient} for interating with document databases.
+     *
+     * The provided runtime will be used for handling object persistence.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DocumentClient client = DocumentClient.create(DynamoDbRuntime.class)) {
+     *         client.listRepositories().repositories().forEach(System.out::println);
+     *     }
+     * </code>
+     */
+    static DocumentClient create(Class<? extends DocumentClientRuntime> runtimeClass);
+
+    /**
+     * Create a {@link DocumentClient.Builder} for configuring and creating {@link DocumentClient}s.
+     *
+     * The provided runtime will be used for handling object persistence.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DocumentClient client = DocumentClient.builder(DynamoDbRuntime.class)
+     *                                                .putOption(DynamoDbOption.CLIENT, DynamoDbClient.create())
+     *                                                .build()) {
+     *         client.listRepositories().repositories().forEach(System.out::println);
+     *     }
+     * </code>
+     */
+    static DocumentClient.Builder builder(Class<? extends DocumentClientRuntime> runtimeClass);
+
+    /**
+     * Get all {@link DocumentRepository}s that are currently available from this client.
+     *
+     * This should return every repository that will not result in {@code client.repository(...)} throwing a
+     * {@link NoSuchRepositoryException}.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DocumentClient client = DocumentClient.create(DynamoDbRuntime.class)) {
+     *         client.listRepositories().repositories().forEach(System.out::println);
+     *     }
+     * </code>
+     */
+    ListRepositoriesResponse listRepositories();
+
+    /**
+     * Retrieve a {@link DocumentRepository} based on the provided repository name/id.
+     *
+     * The {@link DocumentRepository} is used to directly interact with entities in the remote repository.
+     * See {@link #mappedRepository(Class)} for a way of interacting with the repository using Java objects.
+     *
+     * If the repository does not exist, an exception is thrown.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DocumentClient client = DocumentClient.create(DynamoDbRuntime.class)) {
+     *         DocumentRepository repository = client.repository("my-table");
+     *         assert repository.name().equals("my-table");
```

Comment:
Why is the assertion on repository.name() and the lookup on just repository("name")